src/map/status.c

```diff
@@ -3552,39 +3552,20 @@ static void status_calc_regen(struct block_list *bl, struct status_data *st, str
 	}
 }
 
-//Calculates SC related regen rates.
-static void status_calc_regen_rate(struct block_list *bl, struct regen_data *regen, struct status_change *sc)
+static void status_calc_regen_rate_pc(struct map_session_data *sd, struct regen_data *regen)
 {
-	nullpo_retv(bl);
-	if (!(bl->type&BL_REGEN) || !regen)
-		return;
-
-	regen->rate.hp = 100;
-	regen->rate.sp = 100;
-
-	if (regen->skill != NULL) {
-		regen->skill->rate.hp = 100;
-		regen->skill->rate.sp = 100;
-	}
-
-	if (regen->sitting != NULL) {
-		regen->sitting->rate.hp = 100;
-		regen->sitting->rate.sp = 100;
-	}
-
-	if (bl->type == BL_PC) {
-		struct map_session_data *sd = BL_UCAST(BL_PC, bl);
-		struct guild_castle *gc = guild->mapindex2gc(bl->m);
+	nullpo_retv(sd);
+	nullpo_retv(regen);
+	nullpo_retv(regen->skill);
```

Comment:
is it safe to return here? can't a player have only one of the two?

Reply:
if a pc object have any of those null it should trigger an assert, it should never happen. (and it would not with current code anyway)

Reply:
alright